### PR TITLE
Replace `sleep` with `wait_for_file`

### LINKED ISSUE #2444: Replace sleep with wait_for_file()
### What new functionality do you need?

In many places, our code waits for a file to exist by repeatedly checking its existence, then sleeping for a period if it is not found before checking again or until a maximum number of attempts are made. This behavior is repeated across multiple scripts and disrupts the flow of the script to do this checking. While these should be removed as much as possible to prevent wasting resources, this functionality will still be needed in some instances.

PR #2374 introduced a new bash function, `wait_for_file()` that handles the details of waiting for a file. That PR replaced the relevant instances in gempak scripts, but those elsewhere still need to be replaced.

### What are the requirements for the new functionality?

All instances where bash scripts are waiting for a file replaced with the `wait_for_file` bash function.

### Acceptance Criteria

- [ ] Any instance of waiting for a file replaced with the `wait_for_file()` function (no `sleep` elsewhere)
- [ ] Output identical to existing

### Suggest a solution (optional)

`wait_for_file()` is best used in an `if` block as follows:

```
if ! wait_for_file ${filename} ${sleep_interval} ${max_tries}; then
    <code if file is still not available, usually a fatal error>
fi
```

Note the negation and direct use of the function without using `[[ ]]`.

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
# Description

This PR addresses issue #2444. The following is accomplished:

- All `sleep` statements are replaced with `wait_for_file` for the relevant scripts beneath `scripts` and `ush`;
- Indentation and shell-norms are updated where applicable. 

Note: The WAFS scripts are not updated as per @aerorahul direction.

  Resolves #2444 

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
The following CI tests were executed for RDHPCS Hera:

- `C96_atm3DVar`
- `C48_S2SW`

The remaining configurations will be evaluated via the full CI/CD tests.

# Checklist
- [x] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA This passes the the CI/CD tests. However, the shell linter remains an issue.

File: scripts/exgfs_atmos_postsnd.sh [hunk 1]
```diff
@@ -50,6 +50,7 @@ declare -x LEVS
 
 ### Loop for the hour and wait for the sigma and surface flux file:
 export FSTART=$STARTHOUR
+sleep_interval=10
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_atmos_postsnd.sh:53]:
```suggestion
sleep_interval=10
max_tries=360
```

File: scripts/exgfs_atmos_postsnd.sh [hunk 2]
```diff
@@ -69,20 +70,10 @@ export FINT=$NINT1
...
-          break
-      fi
-
-      if [ $ic -ge 360 ]
-      then
-         err_exit "COULD NOT LOCATE logf$FEND file AFTER 1 HOUR"
-      fi
-   done
+   filename="${COM_ATMOS_HISTORY}/${RUN}.${cycle}.atm.logf${FEND}.${logfm}"
+   if ! wait_for_file "${filename}" "${sleep_interval}" "360"; then
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_atmos_postsnd.sh]:
```suggestion
   if ! wait_for_file "${filename}" "${sleep_interval}" "${max_tries}"; then
```

File: scripts/exgfs_atmos_postsnd.sh [hunk 3]
```diff
@@ -69,20 +70,10 @@ export FINT=$NINT1
...
+     err_exit "COULD NOT LOCATE logf${FEND} file"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_atmos_postsnd.sh]:
```suggestion
     err_exit "FATAL ERROR: logf${FEND} not found after waiting $((sleep_interval * ( max_tries - 1))) secs"
```

File: scripts/exgfs_wave_nawips.sh [hunk 4]
```diff
@@ -71,29 +71,19 @@ while [ ${fhcnt} -le ${FHMAX_WAV} ]; do
...
-        echo '*** ERROR : NO GRIB FILE *** '
-        echo '**************************** '
-        echo ' '
-        echo ${msg}
-        set_trace
-        echo "${RUNwave} ${grdID} ${fhr} prdgen ${date} ${cycle} : GRIB file missing." >> ${wavelog}
-        err=1;export err;${errchk} || exit ${err}
-      fi
-    done
+    sleep_interval=20
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_nawips.sh]: Move this up next to where maxtries is defined.

File: scripts/exgfs_wave_nawips.sh [hunk 5]
```diff
@@ -71,29 +71,19 @@ while [ ${fhcnt} -le ${FHMAX_WAV} ]; do
...
+      msg="ABORTING waiting for ${GRIBIN}."
+      echo ' '
+      echo '**************************** '
+      echo '*** ERROR : NO GRIB FILE *** '
+      echo '**************************** '
+      echo ' '
+      echo "${msg}"
+      set_trace
+      echo "${RUNwave} ${grdID} ${fhr} prdgen ${date} ${cycle} : GRIB file missing." >> "${wavelog}"
+      err=1;export err;"${errchk}" || exit "${err}"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_nawips.sh]:
```suggestion
      echo "FATAL ERROR: ${GRIBIN_chk} not found after waiting $((sleep_interval * ( max_tries - 1))) secs"
      echo "${RUNwave} ${grdID} ${fhr} prdgen ${date} ${cycle} : GRIB file missing." >> "${wavelog}"
      err=1;export err;"${errchk}" || exit "${err}"
```

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 6]
```diff
@@ -252,26 +252,22 @@ source "${USHgfs}/preamble.sh"
...
-          echo ' '
-          set_trace
-          echo "${WAV_MOD_TAG} post ${grdID} ${PDY} ${cycle} : field output missing."
-          err=3; export err;${errchk}
-          exit $err
-        fi
-        ${NLN} ${gfile} ./out_grd.${wavGRD}
-      done
+    # Gridded data (main part, need to be run side-by-side with forecast
+    sleep_interval=10
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh]: Move this up next to iwaitmax

File: scripts/exgfs_wave_nawips.sh [hunk 7]
```diff
@@ -71,29 +71,19 @@ while [ ${fhcnt} -le ${FHMAX_WAV} ]; do
...
-        echo '**************************** '
-        echo ' '
-        echo ${msg}
-        set_trace
-        echo "${RUNwave} ${grdID} ${fhr} prdgen ${date} ${cycle} : GRIB file missing." >> ${wavelog}
-        err=1;export err;${errchk} || exit ${err}
-      fi
-    done
+    sleep_interval=20
+    if ! wait_for_file "${GRIBIN}" "${sleep_interval}" "${maxtries}"; then
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_nawips.sh]: It's important to key off of the index file so processing doesn't start prematurely.
```suggestion
    GRIBIN_chk="${GRIBIN}.idx"
    if ! wait_for_file "${GRIBIN_chk}" "${sleep_interval}" "${maxtries}"; then
```

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 8]
```diff
@@ -252,26 +252,22 @@ source "${USHgfs}/preamble.sh"
     export GRIBDATA=${DATA}/output_$YMDHMS
     export GRDIDATA=${DATA}/output_$YMDHMS
 
-# Gridded data (main part, need to be run side-by-side with forecast
-
-    if [ $fhr = $fhrg ]
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh:257]: What happened to this `if`?

File: scripts/exgfs_wave_prdgen_gridded.sh [hunk 9]
```diff
@@ -96,30 +96,19 @@ grids=${grids:-ak_10m at_10m ep_10m wc_10m glo_30m}
...
-         echo '**************************** '
-         echo ' '
-         echo $msg
-         set_trace
-         echo "$RUNwave $grdID ${fhr} prdgen $date $cycle : GRIB file missing." >> $wavelog
-         err=1;export err;${errchk} || exit ${err}
-       fi
-     done
+     sleep_interval=5
+     if ! wait_for_file "${GRIBIN}" "${sleep_interval}" "1000"; then
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_prdgen_gridded.sh]:
```suggestion
     max_tries=1000
     if ! wait_for_file "${GRIBIN_chk}" "${sleep_interval}" "${max_tries}"; then
```

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 10]
```diff
@@ -252,26 +252,22 @@ source "${USHgfs}/preamble.sh"
...
+      gfile="${COM_WAVE_HISTORY}/${WAV_MOD_TAG}.out_grd.${wavGRD}.${YMD}.${HMS}"
+      if ! wait_for_file "${gfile}" "${sleep_interval}" "${iwaitmax}"; then
+        echo '*************************************************** '
+        echo " FATAL ERROR : NO RAW FIELD OUTPUT FILE out_grd.${grdID} "
+        echo '*************************************************** '
+        echo ' '
+        set_trace
+        echo "${WAV_MOD_TAG} post ${grdID} ${PDY} ${cycle} : field output missing."
+        err=3; export err; "${errchk}"
+        exit "${err}"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh]:
```suggestion
        echo " FATAL ERROR : NO RAW FIELD OUTPUT FILE out_grd.${grdID}"
        echo "${WAV_MOD_TAG} post ${grdID} ${PDY} ${cycle} : field output missing."
        err=3; export err; "${errchk}"
        exit "${err}"
```

File: scripts/exgfs_wave_prdgen_gridded.sh [hunk 11]
```diff
@@ -96,30 +96,19 @@ grids=${grids:-ak_10m at_10m ep_10m wc_10m glo_30m}
      #
 
      GRIBIN="${COM_WAVE_GRID}/${RUNwave}.${cycle}.${grdID}.f${fhr}.grib2"
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_prdgen_gridded.sh:98]:
```suggestion
     GRIBIN="${COM_WAVE_GRID}/${RUNwave}.${cycle}.${grdID}.f${fhr}.grib2"
     GRIBIN_chk="${GRIBIN}.idx"
```

File: scripts/exgfs_wave_prdgen_gridded.sh [hunk 12]
```diff
@@ -96,30 +96,19 @@ grids=${grids:-ak_10m at_10m ep_10m wc_10m glo_30m}
...
+     sleep_interval=5
+     if ! wait_for_file "${GRIBIN}" "${sleep_interval}" "1000"; then
+       msg="ABNORMAL EXIT: NO GRIB FILE FOR GRID $GRIBIN"
+       echo ' '
+       echo '**************************** '
+       echo '*** ERROR : NO GRIB FILE *** '
+       echo '**************************** '
+       echo ' '
+       echo $msg
+       set_trace
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_prdgen_gridded.sh]:
```suggestion
       echo "FATAL ERROR: ${GRIBIN_chk} not found after waiting $((sleep_interval * ( max_tries - 1))) secs"
```

File: scripts/exgfs_wave_prdgen_gridded.sh [hunk 13]
```diff
@@ -149,15 +138,14 @@ grids=${grids:-ak_10m at_10m ep_10m wc_10m glo_30m}
...
-     GRIBIN_chk=$GRIBIN.idx
 
-     ${NLN} $GRIBIN gribfile.$grdID.f${fhr}
+     ln -s $GRIBIN gribfile.$grdID.f${fhr}
 
      #
 # 1.d Input template files
      parmfile=${PARMgfs}/wave/grib2_${RUNwave}.$grdOut.f${fhr}
      if [ -f $parmfile ]; then
-       ${NLN} $parmfile awipsgrb.$grdID.f${fhr}
```
Comment (WalterKolczynski-NOAA) [inline scripts/exgfs_wave_prdgen_gridded.sh]: Why is any of this changed?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 14]
```diff
@@ -252,26 +252,22 @@ source "${USHgfs}/preamble.sh"
...
+      fi
+      ln -s "${gfile}" "./out_grd.${wavGRD}"
```
Comment (JessicaMeixner-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh]: Are we moving away from ${NLN}?

File: scripts/exgfs_wave_nawips.sh (hunk 4 above)
Comment (HenryRWinterbottom) [reply scripts/exgfs_wave_nawips.sh]: Done.

File: scripts/exgfs_wave_post_gridded_sbs.sh (hunk 14 above)
Comment (HenryRWinterbottom) [reply scripts/exgfs_wave_post_gridded_sbs.sh]: Corrected.

File: scripts/exgfs_wave_post_gridded_sbs.sh (hunk 6 above)
Comment (HenryRWinterbottom) [reply scripts/exgfs_wave_post_gridded_sbs.sh]: Done.

File: scripts/exgfs_wave_prdgen_gridded.sh (hunk 13 above)
Comment (HenryRWinterbottom) [reply scripts/exgfs_wave_prdgen_gridded.sh]: Corrected.

**Comment by HenryRWinterbottom** [comment]
Addressing conflicts now.

**Comment by WalterKolczynski-NOAA** [comment]
@HenryWinterbottom-NOAA Your force-push destroyed all the changes while introducing additional problems.

**Comment by HenryRWinterbottom** [comment]
> @HenryWinterbottom-NOAA Your force-push destroyed all the changes while introducing additional problems.

Are you referring to the conflicts?

File: ush/gfs_bufr_netcdf.sh [hunk 15]
```diff
@@ -49,7 +49,6 @@ else
 fi
 
 SFCF="sfc"
-CLASS="class1fv3"
```
Comment (WalterKolczynski-NOAA) [inline ush/gfs_bufr_netcdf.sh]: Why did you remove this? It is used later on.

File: ush/wave_tar.sh [hunk 16]
```diff
@@ -98,21 +98,33 @@ source "${USHgfs}/preamble.sh"
...
+
+      filename="${ID}.${cycle}.${type}_tar" 
+      if ! wait_for_file "${filename}" "${sleep_interval}" "${countMAX}" ; then
+        set +x
+        echo ' '
+        echo '***************************************** '
+        echo '*** FATAL ERROR : TAR CREATION FAILED *** '
+        echo '***************************************** '
+        echo ' '
+        set_trace
```
Comment (WalterKolczynski-NOAA) [inline ush/wave_tar.sh]:
```suggestion
        echo "FATAL ERROR: File ${filename} not found after waiting $(( sleep_interval * (countMAX + 1) secs"
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: ush/gfs_bufr_netcdf.sh (hunk 15 above)
Comment (HenryRWinterbottom) [reply ush/gfs_bufr_netcdf.sh]: Fixed.

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 17]
```diff
@@ -252,27 +253,20 @@ source "${USHgfs}/preamble.sh"
     export GRIBDATA=${DATA}/output_$YMDHMS
     export GRDIDATA=${DATA}/output_$YMDHMS
 
-# Gridded data (main part, need to be run side-by-side with forecast
```
Comment (JessicaMeixner-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh:255]: Can this comment be added back in, or was it removed for a reason?

Comment (JessicaMeixner-NOAA) [reply scripts/exgfs_wave_post_gridded_sbs.sh:255]: The comment was moved I see not sure why though.

File: ush/wave_tar.sh [hunk 18]
```diff
@@ -98,21 +98,27 @@ source "${USHgfs}/preamble.sh"
...
     
     nf=$(ls | awk '/'$ID.*.$filext'/ {a++} END {print a}')
     nbm2=$(( $nb - 2 ))
-    if [ $nf -ge $nbm2 ]
-    then 
-      tar -cf $ID.$cycle.${type}_tar ./$ID.*.$filext
+    if [[ "${nf}" -ge "${nbm2}" ]]
+    then
+
+      filename="${ID}.${cycle}.${type}_tar"
```
Comment (JessicaMeixner-NOAA) [inline ush/wave_tar.sh:115]: Adding filename here is a great addition, is there a reason we don't use this in other places too?  Lastly, I'm concerned about the change in order here.  Shouldn't we create the tar-file before we wait to see if it exists? I might not have looked at the logic long enough here, but this seems out of order now.

File: scripts/exgfs_wave_post_gridded_sbs.sh (hunk 17 above)
Comment (HenryRWinterbottom) [reply scripts/exgfs_wave_post_gridded_sbs.sh:255]: Fixed.

Comment (JessicaMeixner-NOAA) [reply scripts/exgfs_wave_post_gridded_sbs.sh:255]: Thanks @HenryWinterbottom-NOAA !

File: scripts/exgfs_wave_post_gridded_sbs.sh [hunk 19]
```diff
@@ -253,26 +254,21 @@ source "${USHgfs}/preamble.sh"
...
-      for wavGRD in ${waveGRD} ; do
-        gfile=${COM_WAVE_HISTORY}/${WAV_MOD_TAG}.out_grd.${wavGRD}.${YMD}.${HMS}
-        while [ ! -s ${gfile} ]; do sleep 10; let iwait=iwait+1; done
-        if [ $iwait -eq $iwaitmax ]; then
-          echo '*************************************************** '
-          echo " FATAL ERROR : NO RAW FIELD OUTPUT FILE out_grd.$grdID "
-          echo '*************************************************** '
-          echo ' '
-          set_trace
+      # Gridded data (main part, need to be run side-by-side with forecast
```
Comment (JessicaMeixner-NOAA) [inline scripts/exgfs_wave_post_gridded_sbs.sh]: This comment can now be deleted as it was added back above.  Thanks again for doing that!

File: ush/wave_tar.sh (hunk 18 above)
Comment (WalterKolczynski-NOAA) [reply ush/wave_tar.sh:115]: Yup, the tar creation needs to be before the wait line. But I'm not really sure of the point of how it was set up originally anyway. Shouldn't be waiting on the existence of the files that are going _into_ the tar?

Comment (JessicaMeixner-NOAA) [reply ush/wave_tar.sh:115]: Agreed, it looks a bit odd which is why i was thinking maybe @HenryWinterbottom-NOAA was seeing something I wasn't seeing after a quick look.  What I suspect is happening here is that we're making sure the tar file exists and is a certain age before creating a tar.gz below -- but again a quick look and would have to do a deeper dive to refresh my memory or understand why things are this way.

Comment (WalterKolczynski-NOAA) [reply ush/wave_tar.sh:115]: Yeah. For now, the task is just to replicate the functionality of what was there before (so make the tarball first). When these get refactored, we will need to look at changing it (this job looks ripe for the wxflow `FileHandler`).

Comment (HenryRWinterbottom) [reply ush/wave_tar.sh:115]: Is this resolved or am I missing something here?

Comment (WalterKolczynski-NOAA) [reply ush/wave_tar.sh:115]: @HenryWinterbottom-NOAA not resolved. You need to move the line with the `tar` command above where you check for the resultant file.

Comment (HenryRWinterbottom) [reply ush/wave_tar.sh:115]: Done.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Conditionally approved pending successful completion of CI tests.

**Comment by WalterKolczynski-NOAA** [comment]
There are `*` inside of quotation marks, so they are being treated as literals instead of being converted to globs.

File: ush/wave_tar.sh [hunk 20]
```diff
@@ -124,21 +130,15 @@ source "${USHgfs}/preamble.sh"
         exit 3
       fi
       
-      if [ -f "$ID.$cycle.${type}_tar" ]
+      if [[ -f "${ID}.${cycle}.${type}_tar" ]]
       then
         tardone='yes'
       fi
-    else
```
Comment (JessicaMeixner-NOAA) [inline ush/wave_tar.sh:131]: I see that a lot of the wave post point jobs are failing in CI.  I didn't see the errors but looking again at this code, it looks like the logic here was changed.  Perhaps that's the issue?

**Comment by HenryRWinterbottom** [comment]
I am testing a C48_S2SW on RDHPCS Hera now with the fix. I will follow-up once I have results.

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA I pushed the correction. The fix resulted in the following Rocoto status:

```
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
202103231200             gfsstage_ic                    61084971           SUCCEEDED                   0         1          25.0
202103231200             gfswaveinit                    61084972           SUCCEEDED                   0         1          31.0
202103231200                 gfsfcst                    61085153           SUCCEEDED                   0         1        2269.0
202103231200    gfsatmos_prod_f000-f006                    61085441           SUCCEEDED                   0         1         423.0
202103231200    gfsatmos_prod_f009-f015                    61085605           SUCCEEDED                   0         1         419.0
202103231200    gfsatmos_prod_f018-f024                    61085609           SUCCEEDED                   0         1         415.0
202103231200    gfsatmos_prod_f027-f033                    61085746           SUCCEEDED                   0         1         385.0
202103231200    gfsatmos_prod_f036-f042                    61085745           SUCCEEDED                   0         1         383.0
202103231200    gfsatmos_prod_f045-f051                    61085847           SUCCEEDED                   0         1         408.0
202103231200    gfsatmos_prod_f054-f060                    61085846           SUCCEEDED                   0         1         398.0
202103231200    gfsatmos_prod_f063-f069                    61085989           SUCCEEDED                   0         1         389.0
202103231200    gfsatmos_prod_f072-f078                    61086123           SUCCEEDED                   0         1         394.0
202103231200    gfsatmos_prod_f081-f087                    61086124           SUCCEEDED                   0         1         407.0
202103231200    gfsatmos_prod_f090-f096                    61086362           SUCCEEDED                   0         1         394.0
202103231200    gfsatmos_prod_f099-f105                    61086375           SUCCEEDED                   0         1         396.0
202103231200    gfsatmos_prod_f108-f114                    61086538           SUCCEEDED                   0         1         388.0
202103231200    gfsatmos_prod_f117-f120                    61086539           SUCCEEDED                   0         1         262.0
202103231200              gfstracker                    61087067           SUCCEEDED                   0         1          71.0
202103231200              gfsgenesis                    61087068           SUCCEEDED                   0         1         529.0
202103231200    gfsocean_prod_f006-f018                    61085671           SUCCEEDED                   0         1         117.0
202103231200    gfsocean_prod_f024-f036                    61085768           SUCCEEDED                   0         1          75.0
202103231200    gfsocean_prod_f042-f054                    61085873           SUCCEEDED                   0         1          83.0
202103231200    gfsocean_prod_f060-f072                    61086187           SUCCEEDED                   0         1         119.0
202103231200    gfsocean_prod_f078-f090                    61086382           SUCCEEDED                   0         1          77.0
202103231200    gfsocean_prod_f096-f108                    61086569           SUCCEEDED                   0         1          78.0
202103231200    gfsocean_prod_f114-f120                    61086602           SUCCEEDED                   0         1          55.0
202103231200    gfsice_prod_f006-f018                    61085670           SUCCEEDED                   0         1          98.0
202103231200    gfsice_prod_f024-f036                    61085769           SUCCEEDED                   0         1          57.0
202103231200    gfsice_prod_f042-f054                    61085874           SUCCEEDED                   0         1          68.0
202103231200    gfsice_prod_f060-f072                    61085990           SUCCEEDED                   0         1          68.0
202103231200    gfsice_prod_f078-f090                    61086188           SUCCEEDED                   0         1          64.0
202103231200    gfsice_prod_f096-f108                    61086383           SUCCEEDED                   0         1          61.0
202103231200    gfsice_prod_f114-f120                    61086603           SUCCEEDED                   0         1          39.0
202103231200          gfswavepostsbs                    61085442           SUCCEEDED                   0         1        2161.0
202103231200          gfswavepostpnt                    61086604           SUCCEEDED                   0         1       11147.0
202103231200                 gfsarch                    61094297           SUCCEEDED                   0         1          55.0
202103231200              gfscleanup                    61094490           SUCCEEDED                   0         1          30.0
```

**Comment by WalterKolczynski-NOAA** [comment]
There's an error in the sounding job, which only gets run by the CI on WCOSS2 (but should work on other machines to test manually).

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA Bug fixed. I will test when RDHPCS Hera is back online.

**Comment by WalterKolczynski-NOAA** [comment]
@HenryWinterbottom-NOAA is this ready to test again?

**Comment by HenryRWinterbottom** [comment]
Yes. I corrected the wild card issue. I can do a quick test to make sure it runs to completion on RDHPCS Hera before you launch the full CI tests. Does that work for you?

Thank you for the reminder.

**Comment by HenryRWinterbottom** [comment]
> Yes. I corrected the wild card issue. I can do a quick test to make sure it runs to completion on RDHPCS Hera before you launch the full CI tests. Does that work for you?
> 
> Thank you for the reminder.

Testing now. Please stay tuned.

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA The RDHPCS Hera test passed if you wanted to try again.

**Comment by HenryWinterbottom-NOAA** [comment]
> Experiment C96_atm3DVar_extended_088d124e **FAIL** on Wcoss2 at 06/07/24 07:51:25 AM
> 
> Error logs:
> 
> ```
> /lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2586/RUNTESTS/COMROOT/C96_atm3DVar_extended_088d124e/logs/2021122100/gfspostsnd.log
> ```
> 
> Follow link here to view the contents of the above file(s): [(link)](https://gist.github.com/emcbot/e2b79c22b93c0dbdf333f8f1aa9270d0)

I am not seeing the following on RDHPCS Hera:

`/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2586/global-workflow/scripts/exgfs_atmos_postsnd.sh: line 80: [[: 048: value too great for base (error token is "048")
+ exgfs_atmos_postsnd.sh[83]: /lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2586/global-workflow/ush/gfs_bufr.sh`

I am assuming that it is because this task only runs for WCOSS. Is this pointing to an issue elsewhere and unrelated to the topic of this PR or a lingering bug related to this PR?

**Comment by HenryRWinterbottom** [comment]
Testing this again for RDHPCS Hera to determine what the lingering issue is. I will update this thread once I have the answer.

**Comment by WalterKolczynski-NOAA** [comment]
> Testing this again for RDHPCS Hera to determine what the lingering issue is. I will update this thread once I have the answer.

If hit the wallclock:
```
job killed: walltime 7242 exceeded limit 7200
```

**Comment by WalterKolczynski-NOAA** [comment]
Looks like there is actually at least one error in the script.
```
+ exgfs_atmos_postsnd.sh[80]: [[ 048 -gt 0 ]]
/lfs/h2/emc/global/noscrub/globalworkflow.ci/GFS_CI_ROOT/PR/2586/global-workflow/scripts/exgfs_atmos_postsnd.sh: line 80: [[: 048: value too great for base (error token is "048")
```

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA Is this a bug?

In `parm/gfs/config.base`, I see the following: https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.base#L291

The offending variable, `NEND1`, is defined here:

https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_atmos_postsnd.sh#L42

Thoughts?

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA Is this a bug?
> 
> In `parm/gfs/config.base`, I see the following: https://github.com/NOAA-EMC/global-workflow/blob/develop/parm/config/gfs/config.base#L291
> 
> The offending variable, `NEND1`, is defined here:
> 
> https://github.com/NOAA-EMC/global-workflow/blob/develop/scripts/exgfs_atmos_postsnd.sh#L42
> 
> Thoughts?

The issue isn't in `NEND1`, it is with `FEND`. Please reread the error message and let me know if you don't understand it.

**Comment by HenryRWinterbottom** [comment]
@WalterKolczynski-NOAA Please try the WCOSS2 CI/CD again. The issue has been addressed and runs successfully for RDHPCS Hera.

**Pull request merged**